src/azul/drs.py

```diff
@@ -254,36 +254,49 @@ def access_url(url):
     return {'url': url}
 
 
-class Client:
+@dataclass
+class ClientObject:
```

Comment:
I'm not necessarily objecting to it, but what exactly is gained by bundling all of the arguments into an object? The URL parsing is done here, but that could also be done in a separate method called within `get_object`.

Reply:
The `ClientObject` class idea was born from the need to extract the netloc and `file_uuid` from the `drs_uri` and then pass those values along with the `access_id` and `access_method` variables to the private methods `_get_object()` and `_get_object_access()`. When first approaching this I stored these as instance variables in the `Client` class itself, however it started to seem a bit messy to me to be using `get_object()` as a constructor and method to return a resulting value.
I'll experiment with this again and/or see if just passing the variables to the functions separately is really that bad.

Reply:
> see if just passing the variables to the functions separately

 that's more what I was imagining. I can see how storing them in the client would get messy.